### PR TITLE
An example app for integrating login plugin with zapp login protocols

### PR DESCRIPTION
**Pull request by RoiKed**

### CODE REVIEW

File: LoginPlugin/iOS/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        return retVal
+    }
+    
+	/**
+     This methood is called in order to verify if authorization flow is in process
+    */
+    public func isPerformingAuthorizationFlow() -> Bool {
+        return self.isPerformingAuthorizationFlow()
+    }
+    ```
```
Comment (lomudi) [inline LoginPlugin/iOS/README.md]: you need to remove this space, otherwise, it all will be one big code view.

File: LoginPlugin/iOS/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# ZappLoginPluginExample-iOS
+
+The ZappLoginPluginExample is an example project for creating a login plugin for the Applicaster Zapp Platform. You can use this example project as a reference for how to build your own login plugin.
+
+If you are not familiar with Zapp please visit [our website](http://applicaster.com/?page=product) for more details.
+
+The full [Zapp](http://zapp.applicaster.com) plugins documentation is available [here](http://zapp-tech-book.herokuapp.com).
```
Comment (lomudi) [inline LoginPlugin/iOS/README.md]: please change 'http://zapp-tech-book.herokuapp.com' to 'http://developer-zapp.applicaster.com/'

File: LoginPlugin/iOS/README.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+     Returns bool value indicating if the user is already verified if not  we start the login proccess
+     in this example we check if the login token exists and if it is valid
+    */
+    public func isUserComply(policies:[String: NSObject]) -> Bool {
+        return self.isAuthenticated()
+    }
+```
+
+#### Using the Zapp Plugin Configuration JSON
+When creating a plugin in Zapp we can create custom configuration fields. This enable the Zapp user to fill relevant details for the specific plugin. More details can found in the [Zapp Plugin Manifest Format](http://zapp-tech-book.herokuapp.com/zappifest/plugins-manifest-format.html).
```
Comment (lomudi) [inline LoginPlugin/iOS/README.md]: change the url to 'http://developer-zapp.applicaster.com/zappifest/plugins-manifest-format.html'

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS.podspec [hunk 4]
```diff
@@ -0,0 +1,36 @@
+Pod::Spec.new do |s|
+
+  s.name             = "ZappLoginPluginExample-iOS"
+  s.version          = '1.0.0'
+  s.summary          = "An EXample of video player framework for Zapp iOS."
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS.podspec]: player should be login

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS.podspec [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  s.description      = <<-DESC
+                        An EXample of video player framework for Zapp iOS.
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS.podspec]: player should be login

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS.podspec [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+                       DESC
+  s.homepage         = "https://github.com/applicaster/ZappLoginPluginExample-iOS.git"
+  s.license          = 'MIT'
+  s.author           = { "Udi Lumnitz" => "u.lumnitz@applicaster.com" }
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS.podspec]: i don't remember i wrote this :)

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookInfoViewController.swift [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+//  ZappLoginPlugins
+//
+//  Created by Roi Kedarya on 17/05/2018.
+//  Copyright © 2018 Applicaster. All rights reserved.
+//
+
+import Foundation
+import ZappPlugins
+
+@objc class FacebookInfoViewController: UIViewController {
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookInfoViewController.swift]: can you add a comment with a few words about what this class is doing and its role?

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+//  Copyright © 2018 Applicaster. All rights reserved.
+//
+
+import Foundation
+import ZappPlugins
+import ApplicasterSDK
+import FacebookCore
+import FacebookLogin
+
+@objc class FacebookLoginPlugin: NSObject, ZPLoginProviderProtocol, ZPLoginProviderUserDataProtocol {
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift]: can you add a comment with a few words about what this class is doing and its role?
explain that this is the actual plugin and etc.

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+        super.init()
+    }
+    
+    
+//MARK: ZPLoginProviderUserDataProtocol
+    
+    /**
+     
+    */
+    public func isUserComply(policies:[String: NSObject]) -> Bool {
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift]: missing comment and explanation

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+    }
+    
+    /**
+     Test User in order to login successfuly use these credentials
+     user name: jorge_mtglsje_valdano@tfbnw.net
+     password: applicaster is the best
+    **/
+    
+    /**
+     This method is being called in order to start login process.
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift]: can you explain in which cases this method is called? (in the sdk flow)

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+        if let loginManager = loginManager {
+            loginManager.logOut()
+            completion(.completedSuccessfully)
+        } else {
+            completion(.failed)
+        }
+    }
+    
+    /**
+     This methood is called in order to verify if we need to start a login flow
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift]: can you explain or give an example in which cases it been called?

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/InlineViewController.swift [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+//
+//  Created by Udi Lumnitz on 02/10/2017.
+//  Copyright © 2017 Applicaster. All rights reserved.
+//
+
+import Foundation
+import ZappPlugins
+import ApplicasterSDK
+
+class InlineViewController: UIViewController {
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/InlineViewController.swift]: i think this file should be removed...

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/SelectionScreenViewController.swift [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+//
+//  Created by Roi Kedarya on 17/05/2018.
+//  Copyright © 2018 Applicaster. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import ApplicasterSDK
+
+@objc class SelectionScreenViewController: UIViewController {
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/SelectionScreenViewController.swift]: can you add a comment in a few words what this class is doing?

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/ViewController.swift [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+//
+//  Created by Udi Lumnitz on 19/06/2017.
+//  Copyright © 2017 Applicaster. All rights reserved.
+//
+
+import UIKit
+import ZappPlugins
+import ApplicasterSDK
+
+class ViewController: UIViewController {
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/ViewController.swift]: i think this file can also be removed...

**Review by lomudi — CHANGES_REQUESTED** [review]
thanks, @RoiKed, looks good in general. I added a few comments, mainly regarding documentation.

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/InlineViewController.swift (hunk 12 above)
Comment (RoiKed) [reply LoginPlugin/iOS/ZappLoginPluginExample-iOS/InlineViewController.swift]: yep

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/ViewController.swift (hunk 14 above)
Comment (RoiKed) [reply LoginPlugin/iOS/ZappLoginPluginExample-iOS/ViewController.swift]: yep

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/SelectionScreenViewController.swift (hunk 13 above)
Comment (RoiKed) [reply LoginPlugin/iOS/ZappLoginPluginExample-iOS/SelectionScreenViewController.swift]: done

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift (hunk 10 above)
Comment (RoiKed) [reply LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift]: I added an flow example

**Review by lomudi — CHANGES_REQUESTED** [review]
there is still comments that was not addressed @RoiKed

File: LoginPlugin/iOS/README.md (hunk 1 above)
Comment (RoiKed) [reply LoginPlugin/iOS/README.md]: It is seperated

Comment (lomudi) [reply LoginPlugin/iOS/README.md]: @RoiKed you need to remove the tab before the ```
that how it looks at the moment...
screen shot 2018-05-22 at 15 22 26

Comment (RoiKed) [reply LoginPlugin/iOS/README.md]: screen shot 2018-05-22 at 15 25 19

Comment (lomudi) [reply LoginPlugin/iOS/README.md]: well, @RoiKed ... the GitHub rendering is the one that counts...

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift (hunk 8 above)
Comment (lomudi) [reply LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift]: @RoiKed what about this one?

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+import Foundation
+import ZappPlugins
+import ApplicasterSDK
+import FacebookCore
+import FacebookLogin
+
+/**
+ This is an example for a login plugin imlementation.
+ This shows how to use the ZPLoginProviderProtocol and the ZPLoginProviderUserDataProtocol
+ in order to create and integrate Any login plugin that is needed
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift]: `Any` should be `any`

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+ in this example we choose to implemented the facebook login plugin
+ This is the main calss of the implementation
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/FacebookLoginPlugin.swift]: `calss` should be `class`

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/SelectionScreenViewController.swift [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+//  Created by Roi Kedarya on 17/05/2018.
+//  Copyright © 2018 Applicaster. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import ApplicasterSDK
+
+/*
+ This Vc constist from a list of login plugins ,it starts the app folw form fist phase of selecting and initializing the selected,
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/SelectionScreenViewController.swift]: i guess `folw` should be `flow`

File: LoginPlugin/iOS/ZappLoginPluginExample-iOS/SelectionScreenViewController.swift [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+ same way zappApp invokeds the login plugin
```
Comment (lomudi) [inline LoginPlugin/iOS/ZappLoginPluginExample-iOS/SelectionScreenViewController.swift]: i guess `invokeds` should be `invokes`?

**Review by lomudi — CHANGES_REQUESTED** [review]
@RoiKed, small typos...

**Review by lomudi — APPROVED** [review]
💯

**Pull request merged**